lib/ecto/migration.ex

```diff
@@ -278,6 +281,12 @@ defmodule Ecto.Migration do
       import Ecto.Migration
       @disable_ddl_transaction false
       @before_compile Ecto.Migration
+
+      def after_begin(), do: :ok
+
+      def before_commit(), do: :ok
+
+      defoverridable after_begin: 0, before_commit: 0
```

Comment:
The recommendation in Elixir is to tag them as optional callbacks and do not provide default implementations in favor of `function_exported?` checks.